crates/duckdb/src/lib.rs

```diff
@@ -167,21 +182,44 @@ impl Client {
     /// use stac_duckdb::{Client, Config};
     ///
     /// let config = Config {
-    ///     use_s3_credential_chain: true,
     ///     use_hive_partitioning: true,
     ///     convert_wkb: true,
+    ///     use_s3_credential_chain: true,
+    ///     use_azure_credential_chain: true,
+    ///     install_extensions: true,
+    ///     custom_extension_repository: None,
     /// };
     /// let client = Client::with_config(config);
     /// ```
     pub fn with_config(config: Config) -> Result<Client> {
         let connection = Connection::open_in_memory()?;
-        connection.execute("INSTALL spatial", [])?;
+        if let Some(ref custom_extension_repository) = config.custom_extension_repository {
+            connection.execute(
+                "SET custom_extension_repository = '?'",
+                [custom_extension_repository],
+            )?;
+        }
+        if config.install_extensions {
+            connection.execute("INSTALL spatial", [])?;
+            connection.execute("INSTALL icu", [])?;
+        }
         connection.execute("LOAD spatial", [])?;
-        connection.execute("INSTALL icu", [])?;
         connection.execute("LOAD icu", [])?;
+        if config.use_httpfs && config.install_extensions {
+            connection.execute("INSTALL httpfs", [])?;
+        }
         if config.use_s3_credential_chain {
+            if config.install_extensions {
+                connection.execute("INSTALL aws", [])?;
+            }
```

Comment:
It felt right to only install if we're creating the secret?

Reply:
+1 I think that's right as the docs point out,

>The aws extension adds functionality (e.g., authentication) on top of the httpfs extension's S3 capabilities, using the AWS SDK.